packages/core/src/lib/base_protocol.ts

```diff
@@ -50,6 +50,56 @@ export class BaseProtocol implements IBaseProtocol {
     );
     return peer;
   }
+
+  /**
+   * Retrieves a list of peers based on the specified criteria.
+   *
+   * @param numPeers - The number of peers to retrieve. If 0, all peers are returned.
+   * @param includeBootstrap - If true, includes a bootstrap peer in the result. Useful for protocols like Filter and Store that require only one peer for now.
+   * @returns A Promise that resolves to an array of peers based on the specified criteria.
+   */
+  protected async getPeers({
+    numPeers,
+    includeBootstrap
+  }: {
+    numPeers: number;
+    includeBootstrap: boolean;
+  }): Promise<Peer[]> {
+    // Retrieve all peers that support the protocol
+    const allPeersForProtocol = await getPeersForProtocol(this.peerStore, [
+      this.multicodec
+    ]);
+
+    if (numPeers === 0) {
+      return allPeersForProtocol;
+    }
+
+    // Filter the bootstrap peers if required to include
+    const bootstrapPeers = includeBootstrap
+      ? allPeersForProtocol.filter((peer) => peer.tags.has(Tags.BOOTSTRAP))
+      : [];
+
+    // Filter remaining peers excluding bootstrap peers
+    const remainingPeers = allPeersForProtocol.filter(
+      (peer) => !bootstrapPeers.includes(peer)
+    );
+
+    // Initialize the list of selected peers
+    const selectedPeers: Peer[] = [];
+
+    // Add the bootstrap peers if available and required
+    selectedPeers.push(...bootstrapPeers);
```

Comment:
I'd suggest to have a test just for this function.

if `includeBoostrap: true, numPeers: 1` is returned, you actually return all bootstrap peers.
instead of 
I'd suggest instead to have `maxBootstrapPeers: number` in the signature so you can control how many bootstrap peer you want (probably one).